### PR TITLE
feat: Use uniform_bucket_level_access instead due to deprecation

### PR DESCRIPTION
**Pull request by aaronmak**
Google has deprecated the field `bucket_policy_only` in favor
of `uniform_bucket_level_access` and `bucket_policy_only`
will be removed in the next major release of the provider

Updated project-factory module in setup/main.tf
because v3 doesn't work with GCP provider v3.38.0

Resolves https://github.com/terraform-google-modules/terraform-google-cloud-storage/issues/79

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Review by morgante — CHANGES_REQUESTED** [review]
Thanks for the contribution.

It looks like integration tests are failing:

```
$$$$$$ Running command `terraform output -json` in directory /workspace/test/fixtures/multiple_buckets
multiple_buckets local: Verifying

Profile: multiple_buckets
Version: (not specified)
Target:  local://

  ×  gsutil: gsutil (2 failed)
     ✔  Command: `gsutil ls -p ci-cloud-storage-4332` exit_status should eq 0
     ✔  Command: `gsutil ls -p ci-cloud-storage-4332` stderr should eq ""
     ✔  Command: `gsutil ls -p ci-cloud-storage-4332` stdout should include "multiple-buckets-4j9t-eu-one"
     ✔  Command: `gsutil ls -p ci-cloud-storage-4332` stdout should include "multiple-buckets-4j9t-eu-two"
     ✔  Command: `gsutil ls gs://multiple-buckets-4j9t-eu-two` exit_status should eq 0
     ✔  Command: `gsutil ls gs://multiple-buckets-4j9t-eu-two` stderr should eq ""
     ✔  Command: `gsutil ls gs://multiple-buckets-4j9t-eu-two` stdout should include "gs://multiple-buckets-4j9t-eu-two/dev/"
     ✔  Command: `gsutil ls gs://multiple-buckets-4j9t-eu-two` stdout should include "gs://multiple-buckets-4j9t-eu-two/prod/"
     ✔  Command: `gsutil bucketpolicyonly get gs://multiple-buckets-4j9t-eu-one` exit_status should eq 0
     ✔  Command: `gsutil bucketpolicyonly get gs://multiple-buckets-4j9t-eu-one` stderr should eq ""
     ✔  Command: `gsutil bucketpolicyonly get gs://multiple-buckets-4j9t-eu-one` stdout should include "Enabled: True"
     ✔  Command: `gsutil bucketpolicyonly get gs://multiple-buckets-4j9t-eu-two` exit_status should eq 0
     ✔  Command: `gsutil bucketpolicyonly get gs://multiple-buckets-4j9t-eu-two` stderr should eq ""
     ✔  Command: `gsutil bucketpolicyonly get gs://multiple-buckets-4j9t-eu-two` stdout should include "Enabled: False"
     ✔  Command: `gsutil lifecycle get gs://multiple-buckets-4j9t-eu-one` should eq "NEARLINE"
     ✔  Command: `gsutil lifecycle get gs://multiple-buckets-4j9t-eu-one` should eq "SetStorageClass"
     ✔  Command: `gsutil lifecycle get gs://multiple-buckets-4j9t-eu-one` should eq 10
     ×  Command: `gsutil lifecycle get gs://multiple-buckets-4j9t-eu-one` should eq false
     
     expected: false
          got: nil
     
     (compared using ==)

     ✔  Command: `gsutil lifecycle get gs://multiple-buckets-4j9t-eu-one` should eq ["MULTI_REGIONAL", "STANDARD", "DURABLE_REDUCED_AVAILABILITY"]
     ✔  Command: `gsutil lifecycle get gs://multiple-buckets-4j9t-eu-one` exit_status should eq 0
     ✔  Command: `gsutil lifecycle get gs://multiple-buckets-4j9t-eu-one` stderr should eq ""
     ✔  Command: `gsutil lifecycle get gs://multiple-buckets-4j9t-eu-two` should eq "NEARLINE"
     ✔  Command: `gsutil lifecycle get gs://multiple-buckets-4j9t-eu-two` should eq "SetStorageClass"
     ✔  Command: `gsutil lifecycle get gs://multiple-buckets-4j9t-eu-two` should eq 10
     ×  Command: `gsutil lifecycle get gs://multiple-buckets-4j9t-eu-two` should eq false
```

**Comment by aaronmak** [comment]
Tried to debug but there's only so much I can do since I am unable to create a folder under an org and I cant see the logs from the failed integration test

File: test/setup/main.tf [hunk 1]
```diff
@@ -16,13 +16,15 @@
...
-  random_project_id = "true"
-  org_id            = var.org_id
-  folder_id         = var.folder_id
-  billing_account   = var.billing_account
+  name                      = "ci-cloud-storage"
+  random_project_id         = "true"
+  org_id                    = var.org_id
+  folder_id                 = var.folder_id
+  billing_account           = var.billing_account
+  budget_amount             = var.budget_amount
```
Comment (morgante) [inline test/setup/main.tf:26]: We definitely shouldn't need to set up a budget.

**Review by morgante — CHANGES_REQUESTED** [review]

File: test/setup/main.tf (hunk 1 above)
Comment (aaronmak) [reply test/setup/main.tf:26]: I agree but it seems the project-factory module v7 requires it? https://registry.terraform.io/modules/terraform-google-modules/project-factory/google/7.0.0?tab=inputs

Comment (morgante) [reply test/setup/main.tf:26]: It's optional. You might want to upgrade to the latest version (9.0.0): https://github.com/terraform-google-modules/terraform-google-project-factory

Comment (aaronmak) [reply test/setup/main.tf]: let me try again with v6 which doesn't require it

Comment (morgante) [reply test/setup/main.tf]: To be clear no version has ever required it.https://github.com/terraform-google-modules/terraform-google-project-factory/tree/v7.0.0#inputs

Comment (aaronmak) [reply test/setup/main.tf]: Thanks, i was looking at the wrong thing

**Comment by morgante** [comment]
I think the error is lines 58 and 74 [here](https://github.com/terraform-google-modules/terraform-google-cloud-storage/blob/master/test/integration/multiple_buckets/controls/gsutil.rb#L58-L74), which you could probably remove entirely.

**Review by morgante — APPROVED** [review]
